impress-org/givewp
# Feature: Add Stripe account selection as a per-form setting

## Description

This PR introduces the ability to specify an account that differs from the default one on a per-form basis. Includes a new step for the migration process.

## Affects

### Payment Gateways Block
- **New Attribute**: Added `stripeUseGlobalDefault` and `stripeAccountId` to store settings for Stripe.
- **Stripe Component Enhancements**: 
  - Added a new panel to the inspector sidebar containing three elements: a toggle, a dropdown, and a notice.
  - "Use global default" toggle: Always visible.
  - "Choose account" dropdown: Visible when `useGlobalDefault` is false.
  - "No default account set" notice: Displayed when no global default account is configured and no per-form account is selected.
- **Additional Actions**: When the Donation Form is saved, we update the `give_stripe_per_form_accounts` and `_give_stripe_default_account` meta fields based on user selections. This way, we leverage existing methods to automatically load the appropriate account for each form.

## Visuals

![CleanShot 2023-08-24 at 22 33 08](https://github.com/impress-org/givewp/assets/3921017/1a1baae8-5c08-404b-a882-d2e4b956c5bd)
_Not having a global default account_

![CleanShot 2023-08-24 at 22 28 14](https://github.com/impress-org/givewp/assets/3921017/ca4b863b-22a8-464b-b476-0a08e84a4c30)
_Having a global default account_

## Testing Instructions

- Configure two or more Stripe accounts setting one of them as default
- Add a new v3 form
- Open the form and check if your default account is the one being presented on Stripe.
Eg.: _By providing your card information, you allow **GiveWP Test** to charge your card for future payments in accordance with their terms._
- Edit the form setting another account manually through the per-form settings
- Open the form and check if the new account is now being presented as in the example above.

## Pre-review Checklist






---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1205367244891340
